carp.common/src/commonMain/kotlin/dk/cachet/carp/common/UUID.kt

```diff
@@ -20,15 +22,22 @@ expect class UUID( stringRepresentation: String )
 
     companion object
     {
+        /**
+         * Parse common [UUID] representations, also allowing upper case hexadecimal digits.
+         *
+         * TODO: It might be useful to allow even more flexible [uuid] entry (e.g., surrounded by curly braces, etc.).
```

Comment:
Just as a note. Parse could allow uuid with or without hyphens (some systems like nginx uses this more compact format) and of course be case insensitive. But I would refrain from adding too much smarts to the parse function like finding uuids in arbitrary strings an such. Imho it cleaner if it behaves more like the value type parse methods or the date ditto.

Reply:
We can always add extra functionality now in a backwards compatible way. That's why I left the TODO. :) {aaa-...} is also fairly common (Microsoft).